---
layout: post
title: "Ideogram.js"
date: 2020-10-04 08:30:46
author: "@vsoch"
annotate_criteria: https://rseng.github.io/software/repository/github/eweitz/ideogram/annotate-criteria/
annotate_taxonomy: https://rseng.github.io/software/repository/github/eweitz/ideogram/annotate-taxonomy/
categories:
- Software
---

Visualization is a powerful tool to understand and explore data, but it's often overlooked. 
In scientific visualization, the best libraries can allow a scientist to bring life to something that we
otherwise might not be able to see. Thus, for the software showcase this week, we are proud to share <a href="https://github.com/eweitz/ideogram" target="_blank">Ideogram.js</a>,
a visualization library in JavaScript that can draw and animate genome-wide datasets.

<br>

![{{ site.baseurl }}/assets/img/posts/showcase/ideogram.png]({{ site.baseurl }}/assets/img/posts/showcase/ideogram.png)

<br>

If you are familiar with the RSEPedia or software, we encourage you to contribute to the [research software encyclopedia](https://rseng.github.io/rse/tutorials/annotation/) and annotate the respository:

<ul>
<li><a href="{{ page.annotate_criteria }}" target="_blank">Annotate the software criteria</a></li>
<li><a href="{{ page.annotate_taxonomy }}" target="_blank">Annotate the software taxonomy</a></li>
</ul>

otherwise, keep reading!

<!--more--> 

 - [What is Ideogram.js](#what-is)
 - [How do I cite it?](#cite)
 - [How do I contribute to the software survey](#contribute)
 - [Where can I learn more?](#learn-more)


<a id="what-is">
## What is Ideogram.js

We obviously can't see chromosomes. But being able to visualize them, and understand context (location, possibly an expression
value) could be a useful thing for a scientist to do. But let's ask the lead developer directly! This is from the project README:

> Ideogram supports drawing and animating genome-wide datasets for [human](https://eweitz.github.io/ideogram/human), [mouse](https://eweitz.github.io/ideogram/mouse), and [many other eukaryotes](https://eweitz.github.io/ideogram/eukaryotes).  The [Ideogram API](https://github.com/eweitz/ideogram/blob/master/api.md) for annotations supports [histograms](https://eweitz.github.io/ideogram/annotations-histogram), [heatmaps](https://eweitz.github.io/ideogram/annotations-heatmap), [overlays](https://eweitz.github.io/ideogram/annotations-overlaid), and points of arbitrary shape and color layered in [tracks](https://eweitz.github.io/ideogram/annotations-tracks). Ideogram can depict haploid, [diploid](https://eweitz.github.io/ideogram/ploidy-basic) or higher ploidy genomes (e.g. plants), as well as aneuploidy, [genetic recombination](https://eweitz.github.io/ideogram/ploidy-recombination), and [homologous features](https://eweitz.github.io/ideogram/homology-basic) between chromosomes. 

> Ideogram can be embedded as a [reusable component](https://github.com/eweitz/ideogram#usage) in any web page or application, and leverages D3.js and SVG to achieve fast, crisp client-side rendering. You can also integrate Ideogram with JavaScript frameworks like [Angular](https://github.com/eweitz/ideogram/tree/master/examples/angular), [React](https://github.com/eweitz/ideogram/tree/master/examples/react), and [Vue](https://github.com/eweitz/ideogram/tree/master/examples/vue), as well as data science platforms like [R](https://github.com/eweitz/ideogram/tree/master/examples/r) and [Jupyter Notebook](https://github.com/eweitz/ideogram/tree/master/examples/jupyter).

[![All human genes](https://raw.githubusercontent.com/eweitz/ideogram/master/examples/vanilla/ideogram_histogram_all_human_genes.png)](https://eweitz.github.io/ideogram/annotations_histogram.html)

The lead developer, Eric Weitz, started working on Ideogram.js in [February 2015](https://github.com/eweitz/ideogram/commit/b615e0f0a06fee451f07f5b8582eb772d1097f28), and has developed it as a personal side project on nights and weekends almost daily since.

#### Why is it useful?

As a developer, it's very appealing to have a base or core library that I'm empowered to use to create custom dashboards or interfaces.
Here is an example of that [provided on the site](https://eweitz.github.io/ideogram/differential-expression) showing differential expression:

![{{ site.baseurl }}/assets/img/posts/showcase/ideogram-tool.png]({{ site.baseurl }}/assets/img/posts/showcase/ideogram-tool.png)

You can also take a look at [live examples](https://eweitz.github.io/ideogram/), or see the [project on GitHub](https://github.com/eweitz/ideogram) for installation, usage, and the API.


<a id="cite">
## How do I cite it?

The lead developer recommends the following two citations. The first is not bibtex, but a general
reference to the repository:

```
Weitz, Eric M. Ideogram.js. 2015. https://github.com/eweitz/ideogram.
```

the second is related work that uses Ideogram.js.

```
@Article{ 10.12688/f1000research.9762.1,
AUTHOR = { Weitz, EM and Pantano, L and Zhu, J and Upton, B and Busby, B},
TITLE = {Viewing RNA-seq data on the entire human genome [version 1; peer review: 3 approved]
},
JOURNAL = {F1000Research},
VOLUME = {6},
YEAR = {2017},
NUMBER = {596},
DOI = {10.12688/f1000research.9762.1}
}
```

You can view the paper [here](https://f1000research.com/articles/6-596/v1).

<a id="getting-started">
## How do I get started?
 
 - [Ideogram.js Documentation](https://eweitz.github.io/ideogram/) served via GitHub pages on the site.
 - [Ideogram.js Slides](https://speakerdeck.com/eweitz/designing-genome-visualizations-with-ideogramjs) "Visualizations with Ideogram.js"
 - [Ideogram.js on GitHub](https://github.com/eweitz/ideogram/) 
 - [Ideogram.js: Chromosome visualization with JavaScript](https://speakerdeck.com/eweitz/ideogramjs-chromosome-visualization-with-javascript) October 2018
 - [Designing genome visualizations with Ideogram.js](https://speakerdeck.com/eweitz/designing-genome-visualizations-with-ideogramjs), April 2020


<a id="contribute">
## How do I contribute to the software survey?

<ul>
  <li><a href="{{ page.annotate_criteria }}" target="_blank">Annotate the software criteria</a></li>
  <li><a href="{{ page.annotate_taxonomy }}" target="_blank">Annotate the software taxonomy</a></li>
</ul>

or read more about annotation [here]({{ site.baseurl }}/tutorials/annotate-your-software). You can clone the software repository to do
bulk annotation, or annotation any repository in the <a href="https://rseng.github.io/software/" target="_blank">software database</a>,
We want annotation to be fun, straight-forward, and easy, so we will be showcasing one repository to annotate per week.
If you'd like to request annotation of a particular repository (or addition to the software database)
please don't hesitate to [open an issue](https://github.com/rseng/software/issues) or even a pull request.

<a id="learn-more">
## Where can I learn more?

You might find these other resources useful:

 - [The Research Software Database](https://github.com/rseng/software) on GitHub
 - [RSEpedia Documentation](https://rseng.github.io/rse)
 - [Google Docs Manuscript](https://docs.google.com/document/d/1wDb0udH9OrFWrMBsAVb8RrUMCKKRHoyEep7yveJ1d0k/edit) you are invited to contribute to.
 - [Annotation Documentation for RSEpedia](https://rseng.github.io/rse/tutorials/annotation/)
 - [Annotation Tutorial in RSEng docs](https://rseng.github.io/rse/tutorials/annotation/)

For any resource, you are encouraged to give feedback and contribute!
